node/task.ts

```diff
@@ -17,6 +17,11 @@ export enum TaskResult {
     Failed = 1
 }
 
+/** Hash table of variable info. Only the 'name' and 'secret' properties are populated
+ *  on each VariableInfo.
+ */
+var variableInfo: { [key: string]: VariableInfo; } = { };
```

Comment:
nit - it's not a variableInfo, it's a map of variableInfos.  Also, we should switch to using typescript let.  So?

```
let variableInfoMap: ...
```